packages/uikit/src/molecules/Modal/index.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useEffect } from 'react';
+import { useState } from 'react';
+import ReactDOM from 'react-dom';
+import styled, { css } from '../../lib/styledComponents';
+
+import { LinkButton as Button } from '../../';
+
+export interface UseModalProps {
+  isShowing: boolean;
+  hide: Function;
+}
+
+export const useModal = (): [UseModalProps, () => void] => {
```

Comment:
should we create a `hooks` folder? we might be creating more and then we can easily add tests to them as well?